### ISSUE TITLE
You should open the dialogues as many times as you close them

### ISSUE THREAD

**Issue by yeKcim** [rjl20/hpmor#40]
some chapters have more (or less) **“** than **”** (002, 005, 006, 007, 008,009, 015, 017, 018, 019, 021, 022, …, 062, 064, 071, 072, 075, 077, 078, 079,…)

I checked for { and }, nothing to report

I used this command:
```sh
for f in *.tex; do {
    printf "$f\n " ;
    grep -o "“" $f | wc -l ;
    grep -o "”" $f | wc -l ;
    printf "\n"
}; done
```

**Comment by rrthomas** [comment]
Thanks very much for this! Unfortunately, it's not as simple as "unequal = wrong, equal = right"; for example, in English, you open a quotation mark for each new paragraph of dialogue, but only close it when the character finishes speaking, so if the character speaks multiple paragraphs, there will be more open than close quotation marks.

So at least it is suspicious when there are more closes than opens!

Also, it's not *that* likely that the number will be balanced across a chapter if there are mistakes. So, I revised your script to:

```sh
for f in *.tex; do {
 open=$(grep -o “ $f | wc -l);
 close=$(grep -o ” $f | wc -l);
 if test $open != $close; then printf "$f $open $close\n"; fi;
} done
```

The results are:

```
hpmor-chapter-002.tex 64 66
hpmor-chapter-005.tex 112 116
hpmor-chapter-006.tex 286 288
hpmor-chapter-007.tex 341 345
hpmor-chapter-008.tex 171 175
hpmor-chapter-009.tex 40 42
hpmor-chapter-015.tex 98 100
hpmor-chapter-017.tex 427 433
hpmor-chapter-018.tex 302 304
hpmor-chapter-019.tex 263 265
hpmor-chapter-021.tex 151 155
hpmor-chapter-027.tex 263 265
hpmor-chapter-028.tex 196 198
hpmor-chapter-035.tex 105 107
hpmor-chapter-042.tex 118 120
hpmor-chapter-043.tex 131 133
hpmor-chapter-046.tex 119 121
hpmor-chapter-047.tex 248 252
hpmor-chapter-049.tex 119 123
hpmor-chapter-050.tex 88 90
hpmor-chapter-051.tex 57 59
hpmor-chapter-054.tex 85 89
hpmor-chapter-055.tex 60 62
hpmor-chapter-061.tex 201 203
hpmor-chapter-062.tex 104 102
hpmor-chapter-064.tex 91 89
hpmor-chapter-071.tex 115 117
hpmor-chapter-072.tex 190 192
hpmor-chapter-077.tex 241 243
hpmor-chapter-078.tex 234 238
hpmor-chapter-079.tex 249 251
hpmor-chapter-086.tex 604 606
hpmor-chapter-092.tex 43 45
hpmor-chapter-094.tex 109 111
hpmor-chapter-098.tex 69 71
hpmor-chapter-101.tex 79 81
hpmor-chapter-107.tex 98 100
hpmor-chapter-108.tex 274 280
hpmor-chapter-109.tex 105 107
hpmor-chapter-119.tex 219 221
```

I'll investigate further when I have a minute, starting with where there there are more close quotes than open.

**Comment by yeKcim** [comment]
in French, you only open («) without having closed (») the previous one for long dialogues making several paragraphs. In the first 14 chapters, it didn't seem to me to be the case (the dialogues are rather fast). It's with this script that I started my search in my FR fork (I also had errors with {}), when I saw the same problem for EN I preferred to share in case it could help.

Your script is really better than mine 😁
